### PR TITLE
General inquiry for custom tour.

### PR DESCRIPTION
**Pull request by ablonewolf**

### CODE REVIEW

File: src/main/java/com/technonext/ota/b2c/tour/constant/APIEndpointConstants.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ private APIEndpointConstants() {
     public static final String TOUR_PACKAGE_CONTENT_ENDPOINT= API_VERSION + "/tour-package-content";
     public static final String PACKAGE_CATEGORY_ENDPOINT= API_VERSION + "/package-category";
     public static final String PONT_OF_INTEREST_CONTENT_ENDPOINT = API_VERSION + "/point-of-interest-content";
+    public static final String TOUR_INQUIRY_ENDPOINT= API_VERSION + "/tour/inquiry";
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/constant/APIEndpointConstants.java]: change it to like: tour-inquiry

File: src/main/java/com/technonext/ota/b2c/tour/constant/CustomAPIResponse.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@AllArgsConstructor
+@Getter
+public class CustomAPIResponse<T> {
+    private final HttpStatus status;
+    private final String message;
+    private final T content;
+
+    public CustomAPIResponse(T content) {
+        this.status = HttpStatus.OK;
+        this.message = "Successfully Done!";
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/constant/CustomAPIResponse.java]: in this approach response message will be same for all response as message is not being passed as parameter. so change it and pass message in constructor

File: src/main/java/com/technonext/ota/b2c/tour/model/enums/InquiryFor.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.technonext.ota.b2c.tour.model.enums;
+
+import lombok.Getter;
+
+@Getter
+public enum InquiryFor {
+    Tour_Packages("Tour Packages"), Umrah_Packages("Hajj-Umrah Packages");
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/model/enums/InquiryFor.java]: make it all caps

File: src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+public class TourInquiryServiceImpl implements TourInquiryService {
+    private final TourInquiryRepository tourInquiryRepository;
+
+    @Override
+    public void createCustomTourRequest(CustomTourInquiryRequest tourInquiryRequest) {
+        TourInquiry tourInquiry = TourInquiry.builder()
+            .inquiryChannel(InquiryChannel.WEB)
+            .inquiryType(InquiryType.GENERAL)
+            .inquiryFor(InquiryFor.Tour_Packages)
+            .name(tourInquiryRequest.firstName() + " " + tourInquiryRequest.lastName())
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java]: why need to concat firstName, lastName from web you get only one value

File: src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    String departure,
+    @NotNull
+    LocalDateTime date,
+    String requirements,
+    @NotNull
+    String firstName,
+    @NotNull
+    String lastName,
+    @NotNull
+    String email,
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java:25]: Validated it

File: src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    @NotNull
+    String contact
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java:27]: Validated it

File: src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            .dateOfInquiry(LocalDateTime.now())
+            .preferredJourneyDate(tourInquiryRequest.date())
+            .email(tourInquiryRequest.email())
+            .mobileNumber(tourInquiryRequest.contact())
+            .requirement(tourInquiryRequest.requirements())
+            .inquiryLocation(tourInquiryRequest.destination())
+            .customerLocation(tourInquiryRequest.departure())
+            .inquiryStatus(InquiryStatus.NEW)
+            .b2cUserId(1L)
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java:45]: add TO DO for hard coded userId

File: src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java (hunk 4 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java]: I will finish it later. For the time being, accept this.

File: src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java [hunk 8]
```diff
@@ -8,20 +8,25 @@
...
 @Service
 @RequiredArgsConstructor
 public class TourInquiryServiceImpl implements TourInquiryService {
     private final TourInquiryRepository tourInquiryRepository;
+    private final CommonUtil commonUtil;
 
     @Override
     public void createCustomTourRequest(CustomTourInquiryRequest tourInquiryRequest) {
+        Long countOnDateOfInquiry = tourInquiryRepository.countByDateOfInquiryAfter(LocalDate.now().atStartOfDay());
+        String inquiryNumber = commonUtil.generateUniqueNumber("TCI", countOnDateOfInquiry + 1, 5);
```
Comment (newaz08) [inline src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java]: remove hard coded from java class. use from constant class

File: src/main/java/com/technonext/ota/b2c/tour/constant/APIEndpointConstants.java (hunk 1 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/constant/APIEndpointConstants.java]: Done.

File: src/main/java/com/technonext/ota/b2c/tour/constant/CustomAPIResponse.java (hunk 2 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/constant/CustomAPIResponse.java]: Done.

File: src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java (hunk 5 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java:25]: Done.

File: src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java (hunk 6 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/dto/request/CustomTourInquiryRequest.java:27]: Done.

File: src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java (hunk 7 above)
Comment (ablonewolf) [reply src/main/java/com/technonext/ota/b2c/tour/service/impl/TourInquiryServiceImpl.java:45]: Done.

**Pull request merged**